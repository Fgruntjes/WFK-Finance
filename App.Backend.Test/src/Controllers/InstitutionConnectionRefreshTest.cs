using Microsoft.EntityFrameworkCore;
using Xunit.Sdk;

namespace App.Backend.Test.Controllers;

public class InstitutionConnectionRefreshTest : IClassFixture<InstitutionConnectionRefreshFixture>
{
    private readonly InstitutionConnectionRefreshFixture _fixture;

    public InstitutionConnectionRefreshTest(InstitutionConnectionRefreshFixture fixture)
    {
        _fixture = fixture;
    }

    [Fact]
    public async Task ByExternalId_Success()
    {
        // Act
        var result = await _fixture.Server.ExecuteQuery(new
        {
            _fixture.InstitutionConnectionEntity.ExternalId
        });

        // Assert
        result.MatchSnapshot();
    }

    [Fact]
    public async Task ByExternalId_MissingConnection()
    {
        // Act
        var result = await _fixture.Server.ExecuteQuery(new
        {
            ExternalId = "SomeExternalIdMissing"
        });

        // Assert
        result.MatchSnapshot(m => m.IgnoreField("errors[0].extensions.timestamp"));
    }

    [Fact]
    public async Task ById_Success()
    {
        // Act
        var result = await _fixture.Server.ExecuteQuery(new
        {
            _fixture.InstitutionConnectionEntity.Id
        });

        // Assert response
        result.MatchSnapshot();

        // Assert database
        _fixture.WithData(context =>
        {
            context.InstitutionConnections
                .Include(e => e.Accounts)
                .First(e => e.Id == _fixture.InstitutionConnectionEntity.Id)
                .Accounts.Should().HaveCount(2);
        });
    }

    [Fact]
    public async Task ById_MissingConnection()
    {
        // Act
        var result = await _fixture.Server.ExecuteQuery(new
        {
            Id = new Guid("59a35c45-6e8d-4dc7-bacc-f151f94da93d")
        });

        // Assert
        result.MatchSnapshot(m => m.IgnoreField("errors[0].extensions.timestamp"));
    }

    [Fact]
    public async Task OnlyWithinOrganisation()
    {
        FailException.ForFailure("Not implemented");
    }
}