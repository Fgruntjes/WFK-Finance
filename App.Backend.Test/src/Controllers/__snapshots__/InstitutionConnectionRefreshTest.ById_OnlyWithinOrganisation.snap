{
  "errors": [
    {
      "message": "Connection not found",
      "locations": [
        {
          "line": 3,
          "column": 3
        }
      ],
      "path": [
        "institutionConnection",
        "refreshId"
      ],
      "extensions": {
        "code": "BAD_REQUEST",
        "timestamp": "2023-11-30T11:38:05.619+01:00",
        "severity": "CRITICAL"
      }
    }
  ],
  "data": {
    "institutionConnection": {
      "refreshId": null
    }
  }
}
